seq2seq/adapters/adapter_modeling.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Implements an Adapter Layer."""
+import torch.nn as nn
+
+from .adapter_utils import Activations
+
+
+class Adapter(nn.Module):
+  def __init__(self, model_config, adapter_config):
+    super().__init__()
+    self.input_size = model_config.d_model
+    self.add_layer_norm_after_adapter = adapter_config.add_layer_norm_after_adapter
+    self.weight_init_range = adapter_config.weight_init_range
+    # If reduction factor is not passed we consider default value of 2.
+    reduction_factor = adapter_config.reduction_factor if adapter_config.reduction_factor is not None else 2
```

Comment:
reduction_factor is already defaulted to 16 in AdapterConfig. You may also want an explicit divisibility check for input_size and reduction_factor.

Reply:
done, thanks